File: 04-estructuras-datos/07-tablas-hash/java/ejemplos/HashMapBasico.java
import java.util.HashMap;
import java.util.Map;

public class HashMapBasico {
    public static void main(String[] args) {
        Map<String, Integer> intentos = new HashMap<>();

        // HashMap calcula una ubicación interna a partir del hash de la clave.
        intentos.put("ana", 2);
        intentos.put("luis", 1);

        // get consulta el valor asociado a la clave sin recorrer explícitamente toda la colección.
        System.out.println("Intentos de ana: " + intentos.get("ana"));
    }
}

File: 04-estructuras-datos/07-tablas-hash/scala/ejemplos/HashMapBasico.scala
import scala.collection.mutable.HashMap

object HashMapBasico:
  def main(args: Array[String]): Unit =
    val intentos = HashMap("ana" -> 2, "luis" -> 1)
    // HashMap usa el hash de la clave para organizar el acceso interno.
    println(intentos("ana"))
